src/Components/Components/src/RenderTree/EventArgsTypeCache.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Components.RenderTree
+{
+    internal static class EventArgsTypeCache
+    {
+        private static ConcurrentDictionary<MethodInfo, Type> Cache = new ConcurrentDictionary<MethodInfo, Type>();
```

Comment:
Do we need a concurrent dictionary here? Or are all uses within the sync context? The reason that I'm asking is because this hurts "linkability" in the wasm cases, and I believe there are a bunch of cases where we are using it and we don't need it.

Reply:
It's shared across all circuits